packages/app/src/api/useApiData.ts

```diff
@@ -58,15 +58,16 @@ const localCache: {
  */
 export const useApiData: <T>(
   options: ApiDataHookOptions
-) => { results: T; refreshData(): (() => void) | undefined } = ({
-  noAutoFetch,
-  accessToken,
-  url,
-  headers,
-}) => {
+) => {
+  results: T;
+  refreshData(): (() => void) | undefined;
+  state: undefined | Error | "done";
+} = ({ noAutoFetch, accessToken, url, headers }) => {
   const [results, setResults] = React.useState<any>({});
+  const [state, setState] = React.useState<undefined | Error | "done">();
   const prefixedUrl = usePrefixedUrl(url);
   const refreshData = React.useCallback(() => {
+    setState(undefined);
     if (!accessToken || !prefixedUrl) {
       setResults({});
```

Comment:
should you set state to done here?

Reply:
No, because we are not done if you didnt provide the proper data, but I dont set it as an error because it is just "waiting for you to provide the right stuff". This whole hook needs a rework, but right now if I change that I need to change everything everywhere and I did not want to adress that in this PR.